tests/utils/schema-validator/WorkRequestDotPlot.test.js

```diff
@@ -0,0 +1,76 @@
+const _ = require('lodash');
+const validateRequest = require('../../../src/utils/schema-validator');
+
+describe('Schema validator', () => {
+  it('Validates the DotPlot work request correctly', async () => {
+    const schemaPath = '../../specs/models/work-request-bodies/WorkRequestDotPlot.v1.yaml';
+
+    const validBody = {
+      name: 'DotPlot',
+      markerGenes: true,
+      input: {
+        nGenes: 5,
+        genes: ['Gzma', 'Lyz2'],
+      },
+      subset: {
+        cellClassKey: 'louvain',
+        cellSetKey: 'All',
+      },
+    };
+
+    // It should validate correctly with the correct body
+    const validationResult = await validateRequest(validBody, schemaPath);
+
+    // ValidateRequest does not return anything, so we're expecting undefined
+    expect(validationResult).toBeUndefined();
+
+    // It has to contain the correct name
+    const wrongName = _.merge(validBody, { name: 'WrongName ' });
+
+    // It should throw if it doesn't have the correct name
+    expect(validateRequest(wrongName, schemaPath)).rejects.toThrow();
+
+    // It has to contain markerGenes
+    const noMarkerGenes = _.clone(validBody);
+    delete noMarkerGenes.markerGenes;
+
+    // It should throw if it doesn't have the correct name
+    expect(validateRequest(noMarkerGenes, schemaPath)).rejects.toThrow();
+
+    // It has to contain input
```

Comment:
considering that this comment is related to the `expect` I would remove the space in between preparing the expect and the actual check because they are a single "functional" unit and your comment refers to the last line so the following is easier to understand imho:
```
    // It has to contain input
    const noInput = _.clone(validBody);
    delete noInput.input;
    expect(validateRequest(noInput, schemaPath)).rejects.toThrow();
```

Reply:
updated